### DISCUSSION TITLE
Components not working?

### DISCUSSION THREAD

**Discussion by owzim** [uxbox/uxbox#1316] [Q&A]
I cannot get the components working. If I create a component, and duplicate it and change the shape, other instances of the component do not get updated.

This behavior is known from Figma, Sketch, Illustrator, Photoshop. Is penpot using a drastically different behavior or is it just not working yet?

A gif illustrating what I mean:

![penpot-components](https://user-images.githubusercontent.com/1720088/139935095-3a8e2f15-8f8f-4012-ae7d-11944de18a0f.gif)

**Comment by mlgarchery** [comment]
You have to **right click > Update main components** to see the change be applied to every other components. If you don't do that, a star * is added to the component name in the layer list to show that it differs from the main component.

**Comment by owzim** [comment]
@mlgarchery thanks.

I don't mind the extra step for updating the main component but as you can see the result is very broken. Also as you can see the context menu is not updating correctly at first. Maybe this is more suited for an issue report.

https://user-images.githubusercontent.com/1720088/140211123-de790b69-7f73-49f6-b0e0-c230d2217871.mp4

just for comparison in figma:

https://user-images.githubusercontent.com/1720088/140216321-44db66e4-a055-43b8-bb71-1f8599993f2a.mp4

**Reply by hirunatan** [reply]
I can see in the video that you have duplicated the shape and then you used "Create component" in the copy. This is not necessary, because the copied shape is already a component instance, and you could directly use "Update master" in the shape container. There is a bug in this case, because the app should not allow you to do this. I have created an issue to fix this https://tree.taiga.io/project/penpot/issue/2237

Anyway, the operation of components in Penpot is different than Figma. You can read some documentation about it here: https://help.penpot.app/user-guide/components/
